src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -53,12 +56,360 @@ uint32_t SystemRequest::index = 0;
+    const std::string app_id(app_data[json::appId].asString());
+    if (app_id.length() > max_app_id_length_) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_
+                       << "An Object ID length exceeds maximum allowed ["
+                       << app_id.length() << "]>[" << max_app_id_length_
+                       << "]");
+      result = false;
+    }
+
+    // Verify that appid is unique
+    if (app_ids_map_.find(app_id) != app_ids_map_.end()) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_ << "An Object ID is not unigue [" << app_id
+                                 << "]");
+      result = false;
+    } else {
+      app_ids_map_[app_id] = 1;
+    }
+
+    // Verify that app is not registered yet
+    ApplicationSharedPtr registered_app =
+        ApplicationManagerImpl::instance()->application_by_policy_id(app_id);
+    if (registered_app) {
+      LOG4CXX_DEBUG(logger_,
+                    error_prefix_ << "Application with the same id: " << app_id
+                                  << " is registered already.");
+      result = false;
+    }
+    // Verify app name exist
+    if (!app_data.keyExists(json::name)) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_ << "Can't find app name in json file.");
+      return false;
+    }
+    // And app name length
+    const std::string appName(
+        app_data[json::name].asString());
+    if (appName.length() > max_app_name_length_) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_
+                       << "Name of application exceeds maximum allowed ["
+                       << appName.length() << "]>[" << max_app_name_length_
+                       << "].");
+      result = false;
+    }
+    return result;
+  }
+
+  bool ValidateLanguages(const smart_objects::SmartObject& languages) const {
+    bool result = true;
+    bool default_language_found = false;
+    const size_t languages_array_size = languages.length();
+    if (languages_array_size > max_languages_array_size_) {
+      LOG4CXX_WARN(logger_,
+                   error_prefix_ << "\"languages\" array exceeds max size ["
+                                 << languages_array_size << "]>["
+                                 << max_languages_array_size_ << "]");
+      result = false;
+    }
+    // Every language has ttsname string and vrsynonyms array
+    for (size_t idx = 0; idx < languages_array_size; ++idx) {
+      const smart_objects::SmartObject& language = languages.getElement(idx);
+      std::set<std::string> keys = language.enumerate();
+      for (std::set<std::string>::const_iterator iter = keys.begin();
+           iter != keys.end();
+           ++iter) {
+        // Verify default language defined
+        if (!(*iter).compare(json::default_)) {
+          default_language_found = true;
+        }
+        // Add set for synonyms' duplicates validation
+        if (synonyms_map_.find(*iter) == synonyms_map_.end())
```

Comment:
@okoreniak, same case as with `app_ids_map_`. Give the better name and pass as arg instead of private member.